### ISSUE TITLE
is_countable condition is ignored

### ISSUE THREAD

**Issue by holtkamp** [vimeo/psalm#1742]
I am new to Psalm and maybe this is intended behavior. But since it makes me wonder a bit, I post this issue. 

Used PHP: `7.3.6`
Used Psalm: `3.4.1`

```php

function getObject() : iterable{
   //assemble some iterable
}
$iterableObject = getObject();

if(is_countable($iterableObject)){
   if(count($iterableObject) === 0){
      //do something
   }
   //do other things
}
```

results in a Psalm error like:

```
Argument 1 of count expects array<array-key, mixed>|Countable|SimpleXMLElement|ResourceBundle, iterable<mixed, mixed> provided
            if (count($objects) === 0) {
```

Should / can I expect that Psalm takes the `if_countable` condition into account, or is there a way to instruct Psalm to do this?

**Comment by weirdan** [comment]
`is_countable()` should certainly allow further `count()`. I don't think Psalm has `countable` assertion at the moment though.

**Comment by holtkamp** [comment]
@weirdan ok, thanks for the swift confirmation. So I guess this can be considered as a feature request? 😇

**Comment by weirdan** [comment]
I think so, but don't take it as official response - I'm an ordinary Psalm user, just like you are.

**Comment by muglug** [comment]
If it's good code that Psalm doesn't understand, I generally consider it a bug. This code looks good!

**Comment by muglug** [comment]
a `countable` assertion on `iterable` should turn it into `array|(Traversable&Countable)`

**Comment by weirdan** [comment]
`SimpleXMLElement` does not implement `Countable`, but is countable:
```console
$ php -r 'var_dump(class_implements(SimpleXMLElement::class), is_countable(new SimpleXMLElement("<?xml version=\"1.0\" ?> <a/>")));'
Command line code:1:
array(1) {
  'Traversable' =>
  string(11) "Traversable"
}
Command line code:1:
bool(true)
```

**Comment by weirdan** [comment]
`ResourceBundle` as well:
```console
$ php -r 'var_dump(class_implements(ResourceBundle::class), is_countable(new ResourceBundle("en_GB", null)));'
Command line code:1:
array(1) {
  'Traversable' =>
  string(11) "Traversable"
}
Command line code:1:
bool(true)
```

**Comment by muglug** [comment]
yeah, I'm happy with that bug here (on 7.3 only, as it's fixed in 7.4)
